pkg/trie/triedb/proof/generate.go

```diff
@@ -18,20 +19,22 @@ type nodeHandle interface {
 }
 
 type (
-	nodeHandleHash   common.Hash
+	nodeHandleHash[H any] struct {
```

Comment:
Why not?

```suggestion
	nodeHandleHash[H hash.Hash] struct {
```

Reply:
https://github.com/ChainSafe/gossamer/pull/4221#discussion_r1788099883